spec/currency_spec.rb

```diff
@@ -186,6 +187,27 @@ def to_s
       it "raises UnknownCurrency with unknown currency" do
         expect { Currency.new("xxx") }.to raise_error(Currency::UnknownCurrency, /xxx/)
       end
+
+      it 'returns old object for the same :key' do
+        expect(Currency.new("USD").object_id).to be(Currency.new("USD").object_id)
+        expect(Currency.new("USD").object_id).to be(Currency.new(:usd).object_id)
+        expect(Currency.new("USD").object_id).to be(Currency.new(:USD).object_id)
+        expect(Currency.new("USD").object_id).to be(Currency.new('usd').object_id)
+        expect(Currency.new("USD").object_id).to be(Currency.new('Usd').object_id)
+
+        ids = []
+        [
+          Thread.new{ ids << Currency.new("USD").object_id },
+          Thread.new{ ids << Currency.new("USD").object_id }
+        ].each(&:join)
```

Comment:
For now this one doesn't test anything 'cause usd currency is already in a hash. You should add `Currency.instances.clear` before an example. Also this one seems to be failed in jruby.

To avoid duplication you can use `2.times.map { Thread.new {} }.each(&:join)`.